### PR TITLE
Fixed issue #16633: Error message when changing the data policy message language unless I check the box

### PR DESCRIPTION
**Pull request by gabrieljenik**
Making a form of its own for the language changer of the first page. This form as only have the language will not trigger server side validation of the policy. This is how it is in LS4

### CODE REVIEW

File: themes/survey/vanilla/views/subviews/navigation/language_changer_first_page.twig [hunk 1]
```diff
@@ -17,6 +17,24 @@
...
+    {% set htmlOptions = {
+        'id'          : 'firstpage-changelang',
+        'autocomplete': 'off',
+        'class'       : 'form',
+    } %}
+
+    <!-- Start of language change Form-->
+    {{ C.Html.form(
+        ( aSurveyInfo.surveyUrl ),
+    'post',
```
Comment (Shnoulle) [inline themes/survey/vanilla/views/subviews/navigation/language_changer_first_page.twig:30]: It can be a get here ? Think it's best if it's a get.

**Review by Shnoulle — COMMENTED** [review]
To be checked (and fixed)

- [ ] token by URI or POST value
- [ ] prefilling url https://manual.limesurvey.org/URL_fields#Start_URL
- [ ] captcha

File: themes/survey/vanilla/views/subviews/navigation/language_changer_first_page.twig (hunk 1 above)
Comment (gabrieljenik) [reply themes/survey/vanilla/views/subviews/navigation/language_changer_first_page.twig:30]: Yes, I think so

**Comment by gabrieljenik** [comment]
> To be checked (and fixed)
> 
> * [ ]  token by URI or POST value
> * [ ]  prefilling url https://manual.limesurvey.org/URL_fields#Start_URL
> * [ ]  captcha

This fails when using the language changer from the top menu. Will review again

**Comment by Shnoulle** [comment]
> This fails when using the language changer from the top menu. Will review again

Dis it work currently ? in 3.X and 4.X ?

**Comment by gabrieljenik** [comment]
> This fails when using the language changer from the top menu. Will review again

Now top menu doesn't show error.

**Comment by gabrieljenik** [comment]
> To be checked (and fixed)
> 
> * [ ]  token by URI or POST value
> * [ ]  prefilling url https://manual.limesurvey.org/URL_fields#Start_URL
> * [ ]  captcha

Tested with all. Still sending via POST as the form has all the info, as for example the move parameter, which is used.

**Comment by gabrieljenik** [comment]
> > This fails when using the language changer from the top menu. Will review again
> 
> Dis it work currently ? in 3.X and 4.X ?

This fix is for LS3. I tested in LS3.
On LS4 the error was not reproducible.

**Comment by Shnoulle** [comment]
>  On LS4 the error was not reproducible.

Yes, but i mean : prefill value, token etc ;) .
And to be checked with JS/ without JS too :(

**Comment by gabrieljenik** [comment]
> And to be checked with JS/ without JS too :(

Wihtout JS? Does LS works with no JS support?

**Comment by Shnoulle** [comment]
Public part : yes a lot , i work for it …

**Comment by gabrieljenik** [comment]
> Public part : yes a lot , i work for it …

Haven't tested with no JS support. How do you approach that testing? Any chrome settings could be usefull?

Let me ask, why? JS is a standard!
Isn't that an unnecessary overhead and a limit to development of the tool?

**Comment by Shnoulle** [comment]
> Let me ask, why? JS is a standard!
> Isn't that an unnecessary overhead and a limit to development of the tool?

Not everywhere, and when you connecte with a really light connexion :; it's really great

https://softwareengineering.stackexchange.com/questions/26179/why-do-people-disable-javascript

But the best one is https://softwareengineering.stackexchange.com/a/25996
https://en.wikipedia.org/wiki/Progressive_enhancement

**Comment by gabrieljenik** [comment]
> > Let me ask, why? JS is a standard!
> > Isn't that an unnecessary overhead and a limit to development of the tool?
> 
> Not everywhere, and when you connecte with a really light connexion :; it's really great
> 
> https://softwareengineering.stackexchange.com/questions/26179/why-do-people-disable-javascript
> 
> But the best one is https://softwareengineering.stackexchange.com/a/25996
> https://en.wikipedia.org/wiki/Progressive_enhancement

I see. I like the idea, but is that something to focus right now...?
I will test it..

**Comment by Shnoulle** [comment]
> I see. I like the idea, but is that something to focus right now...?

It's like security … best is to start always by simple part, and improve after.

**Comment by gabrieljenik** [comment]
When testing with no JS the upper menu doesn't ahow up.
Only the language changer from the body. Languages are changed good.

Bug is not reproducible in LS4.
Tested it with no JS as well.

**Comment by Shnoulle** [comment]
Great !

**Pull request merged**